frappe/model/base_document.py

```diff
@@ -185,9 +210,10 @@ def append(self, key, value=None):
 			})
 		"""
 		if value is None:
-			value={}
+			value = {}
+
 		if isinstance(value, (dict, BaseDocument)):
-			if not self.__dict__.get(key):
+			if self.__dict__.get(key) is None:
```

Comment:
Doing this helped fix [places in ERPNext](https://github.com/frappe/erpnext/pull/30634) where different child tables wereinitialised to the same list 😄